src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -935,6 +936,19 @@ def _get_env_and_group_from_log_analytics(
                 env_list = [e for e in env_list if format_location(e["location"]) == format_location(location)]
             if env_list:
                 # TODO check how many CA in env
+                # Get container app list in resource_group
+                containerapp_list = list_containerapp(cmd=cmd, resource_group_name=resource_group_name)
+                if containerapp_list and len(containerapp_list) > 0:
+                    # Get container app in resource_group with same name as app
+                    container_app = next((c for c in containerapp_list if c.get("name", "").lower() == app.name.lower()), None) if app.name else None
+                    if container_app:
+                        # If container app exists, get environmentId and resource_group from container app
+                        logger.info(f"Setting environment for Container App : '{app.name}'")
+                        env_details = parse_resource_id(container_app["properties"]["environmentId"])
+                        # Set environment name and resource_group for the container app
+                        env.set_name(env_details["name"])
```

Comment:
env.set_name should input the environmentId:
```
    def set_name(self, name_or_rid):
        if is_valid_resource_id(name_or_rid):
            env_dict = parse_resource_id(name_or_rid)
            self.name = env_dict["name"]
            self.resource_type = env_dict["resource_type"]
            if "resource_group" in parse_resource_id(name_or_rid):
                rg = parse_resource_id(name_or_rid)["resource_group"]
                if self.resource_group.name != rg:
                    self.resource_group = ResourceGroup(
                        self.cmd,
                        rg,
                        self.location,
                    )
        else:
            self.name = name_or_rid
```

Reply:
Updated.